src/main/java/pl/uj/edu/JImageStream/api/ImageStream.java

```diff
@@ -48,6 +67,17 @@ public ImageStream channel(ColorChannel... colorChannels) {
         return this;
     }
 
+    public ImageStream setThreads(int numberOfThreads) throws Exception{
+        if(!parallel){
+            throw new UnsupportedOperationException("Only parallel streams can use multiple threads");
+        }
+        if(numberOfThreads < 1){
+            throw new Exception("Number of threads must be at least 1");
```

Comment:
don't throw this exception, maybe set default value. We should log some warning here after we implement logging mechanism